android/src/main/java/ly/count/android/sdk/react/CountlyReactNative.java

```diff
@@ -749,6 +749,28 @@ public void event(ReadableArray args) {
         }
     }
 
+    @ReactMethod
+    public void recordEvent(ReadableMap args) {
+        String eventName = args.getString("n");
+
+        int eventCount = 0;
+        if (args.hasKey("c")) {
+            eventCount = args.getInt("c");
+        }
+
+        double eventSum = 0;
+        if (args.hasKey("s")) {
+            eventSum = args.getDouble("s");
+        }
+
+        HashMap<String, Object> segmentation = new HashMap<>();
+        ReadableArray segments = args.getArray("g");
+        for (int i = 0, il = segments != null ? segments.size() : 0; i < il; i += 2) {
+            segmentation.put(segments.getString(i), segments.getString(i + 1));
```

Comment:
although values are Object, they casted to string. Now if a integer passed to here it will be limited by value size.  Value size only meant for strings. There should be a better way to differentiate between types to eliminate this type of anomalies

Reply:
good point

Reply:
I have re-written this.

Reply:
Than you, it is good now. After a little bit tuning ready to go